### PR TITLE
Remove test data from repository and use data from zim-testing-suite.

### PR DESCRIPTION
**Pull request by mgautierfr**
This PR remove the data from the repository and download the data from zim-testing-suite.

It should be considered as a preparatory work of https://github.com/openzim/libzim/pull/535.
It replace https://github.com/openzim/libzim/pull/531 answering comments addressed:
- Put the test data version somewhere in the code to know which data to use with which version of the code.
- Simplify the downloading of the data (now repository contain a scrip to do it and can be run through meson).
- User can download the data and use the downloaded data.

@legoktm On packaging side, you can use the provided script to download the data or make the archive a second source and just configure meson correctly.

For now, this PR works only if `test_data_dir` option is not used (tests look in `test/data` directory in build dir).
This PR must be considered with #535 where test are using data pointed by the env var.
The two PR will be merge together to avoid to add the new zim files to the repository. But I've created a separated PR to concentrate on the download process as we already agreed on the new tests themselves.

### CODE REVIEW

**Review by veloman-yunkan — CHANGES_REQUESTED** [review]
README.md must be updated

**Comment by mgautierfr** [comment]
> README.md must be updated

Done.

File: README.md [hunk 1]
```diff
@@ -107,6 +109,38 @@ If libzim is compiled without Xapian, all search API are removed.  You
...
+By default, meson will use a internal directory in you build directory.
+But you can specify another directory with option `test_data_dir`:
+
+```
+meson . build -Dtest_data_dir=<A_DIR_WITH_TEST_DATA>
+```
+
+Whatever you specify a directory or not, you need a extra step to download the data. At choice:
+- Get the data from the repository [zim-testing-suite](https://github.com/openzim/zim-testing-suite) and put it yourself in the directory.
+- Use the script [download_test_data.py](scripts/download_test_data.py) which will download and extra the data for you.
```
Comment (veloman-yunkan) [inline README.md:126]: extra → extract

File: README.md [hunk 2]
```diff
@@ -107,6 +109,38 @@ If libzim is compiled without Xapian, all search API are removed.  You
 can test if an installed version of libzim is compiled with or without
 xapian by testing the define `LIBZIM_WITH_XAPIAN`.
 
+Testing
+-------
+
+The unit test need testing zim files.
+
+By default, meson will use a internal directory in you build directory.
```
Comment (veloman-yunkan) [inline README.md:117]: you → your

File: README.md [hunk 3]
```diff
@@ -107,6 +109,38 @@ If libzim is compiled without Xapian, all search API are removed.  You
 can test if an installed version of libzim is compiled with or without
 xapian by testing the define `LIBZIM_WITH_XAPIAN`.
 
+Testing
+-------
+
+The unit test need testing zim files.
```
Comment (veloman-yunkan) [inline README.md:115]: Change this sentence to "ZIM files needed by unit-tests are not included in this repository. You must tell meson where the test data is located."

**Review by veloman-yunkan — CHANGES_REQUESTED** [review]

**Review by veloman-yunkan — APPROVED** [review]

**Comment by mgautierfr** [comment]
@legoktm Is it ok for you ?
I don't know how to patch `debian/rules` to add the `ninja download_test_data` during the build.
(I'm not even sure we should as the build step is probably offline.)

**Comment by legoktm** [comment]
> @legoktm Is it ok for you ?

As you had suggested in https://github.com/openzim/libzim/pull/531#issuecomment-819351398, can we have an environment variable that causes it to skip these tests? That can be set in debian/rules and then fixing the packaging no longer blocks you.

> I don't know how to patch `debian/rules` to add the `ninja download_test_data` during the build.
> (I'm not even sure we should as the build step is probably offline.)

Yes, building the Debian package shouldn't require any Internet dependencies. We (I really) should add the packaging configuration to zim-testing-suite so it's in our PPA and libzim can depend on it like a normal apt dependency. I can start working on that over the weekend.

**Comment by kelson42** [comment]
@legoktm Any news about a fix of the deb package CI?

**Comment by mgautierfr** [comment]
@legoktm last commit add a option to deactivate tests using external test data.

We need to pass the option `-Dtest_data_dir=none` to meson.
If you tell we how to pass this option to meson in the debian/rules I can do the patch myself and we will be good.

**Comment by legoktm** [comment]
> @legoktm last commit add a option to deactivate tests using external test data.
> 
> We need to pass the option `-Dtest_data_dir=none` to meson.
> If you tell we how to pass this option to meson in the debian/rules I can do the patch myself and we will be good.

I had to look it up myself :) pushed the fix, which also makes it obvious how to add more config flags in the future.

**Review by legoktm — APPROVED** [review]

**Comment by kelson42** [comment]
There is still two codefactor issues.

**Comment by mgautierfr** [comment]
> There is still two codefactor issues.

I consider them as two false positives.

**Pull request merged**